athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/storage/datasource/StorageMetadataConfig.java

```diff
@@ -0,0 +1,83 @@
+/*-
+ * #%L
+ * athena-hive
+ * %%
+ * Copyright (C) 2019 - 2022 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.amazonaws.athena.connectors.gcs.storage.datasource;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageMetadataConfig
```

Comment:
I could be wrong, but I don't see how this `StorageMetadataConfig` class is used at all?

Its constructed once in `StorageMetadataImpl` but then never used once constructed.

If it is unused, please just delete this entire file and then get rid of its false usage in `StorageMetadataImpl`

Again, as I mentioned in my last review, there are too many classes that are unused in this pull request.

Please get rid of all unused classes, I should not have to keep finding and pointing out these unused classes.

I'm not going to review further as I'm noticing a pattern of unused classes throughout this pull request. Please get rid of all of them (even the ones that I did not cover in my comments).